akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -434,12 +412,12 @@ class TLSActor(
     if (tracing) log.debug("handshake finished")
     val session = engine.getSession
 
-    hostInfo.map(_._1) match {
-      case Some(hostname) if !hostnameVerifier.verify(hostname, session) ⇒
```

Comment:
Ok, but we still cannot remove it now because people may have configured arbitrary HostnameVerifiers with sslconfig. So, maybe just keep it like that for now.